optimum/onnxruntime/io_binding/io_binding_helper.py

```diff
@@ -0,0 +1,170 @@
+class TypeHelper(ORTTypeHelper):
+    """
+    TypeHelper helps get data type information of the ONNX runtime inference session and offers the mapping from
+    OrtValues data type to the data type of other data structures(numpy, torch tensor...).
+    """
+
+    # TODO: Current DLPack doesn't support boolean tensor, use uint8 as workaround, remove after it is supported.
+    @staticmethod
+    def ort_type_to_numpy_type(ort_type: str):
+        ort_type_to_numpy_type_map = {
+            "tensor(int64)": np.int64,
+            "tensor(int32)": np.int32,
+            "tensor(int8)": np.int8,
+            "tensor(float)": np.float32,
+            "tensor(float16)": np.float16,
+            "tensor(bool)": np.uint8,
+        }
+        if ort_type not in ort_type_to_numpy_type_map:
+            raise ValueError(f"{ort_type} not found in map")
+        return ort_type_to_numpy_type_map[ort_type]
+
+    @staticmethod
+    def ort_type_to_torch_type(ort_type: str):
+        ort_type_to_torch_type_map = {
+            "tensor(int64)": torch.int64,
+            "tensor(int32)": torch.int32,
+            "tensor(int8)": torch.int8,
+            "tensor(float)": torch.float32,
+            "tensor(float16)": torch.float16,
+            "tensor(bool)": torch.bool,
+        }
+        if ort_type not in ort_type_to_torch_type_map:
+            raise ValueError(f"{ort_type} not found in map")
+        return ort_type_to_torch_type_map[ort_type]
+
+
+# adapted from https://github.com/microsoft/onnxruntime/blob/1ab11a111ce0717bfbfaca964d04a017cb9b1752/onnxruntime/python/tools/transformers/io_binding_helper.py#L97
+class IOBindingHelper:
+    """
+    IOBindingHelper is a class that helps ORTModels to create buffers for inputs and outputs of an ONNX runtime
+    inference session when using devices like GPU for acceleration. It helps reduce memory copy between the host
+    and device.
+    """
+
+    def __init__(self, model: ort.InferenceSession, config, device, **kwargs):
+        self.model = model
+        self.config = config
+        self.device = device
+        # create {name:idx} dict for model outputs
+        self.model_inputs = {output_key.name: idx for idx, output_key in enumerate(model.get_inputs())}
+        self.model_outputs = {output_key.name: idx for idx, output_key in enumerate(model.get_outputs())}
+        self.model_input_names = list(self.model_inputs.keys())
+        self.model_output_names = list(self.model_outputs.keys())
+
+    def prepare_io_binding(self, **kwargs):
+        """Returnas IO binding object for a session."""
+
+        name_to_np_type = TypeHelper.get_io_numpy_type_map(self.model)
+
+        # Bind inputs and outputs to onnxruntime session
+        io_binding = self.model.io_binding()
+
+        # Bind inputs
+        for input_name in self.model_input_names:
+            onnx_input = kwargs.pop(input_name)
+
+            assert onnx_input.is_contiguous()
```

Comment:
I suggest avoiding `assert` statements and raising a proper error with a helpful error message